### ISSUE TITLE
 CI/CD proposal

### ISSUE THREAD

**Issue by medikoo** [serverless/components#601]
@eahefnawy @ac360  I'd like to setup some CI/CD (with Google actions) In context of this repository.

I propose following:
- Configure ESLint and Prettier with [`@serverless/eslint-config`](https://github.com/serverless/eslint-config)
- Validate ESLint & Prettier formatting on `master` and every PR against master

We may also take into account following additional things:
- Automatic publications of new versions to npm on version change in `package.json`
- Git hooks that validate ESLint and Prettier formatting

Let me know

**Comment by eahefnawy** [comment]
Sounds good 👍 ... I'm still not sold on Git hooks though. Let's just put the basics in place first.

What are Google actions though?! Do you mean Github actions? 🤔

**Comment by medikoo** [comment]
> Sounds good 👍 ... I'm still not sold on Git hooks though. Let's just put the basics in place first.

Yes, I proposed this more for discussion and not for initial setup.

What about though releases made through CD? Can we start with it upfront, or you prefer to eventually upgrade to that later?

> What are Google actions though?! Do you mean Github actions?>

🤦 ofc (updated description)
